pkg/utils/k8sutil.go

```diff
@@ -233,13 +235,29 @@ func GetFunctionData(runtime, ftype, modName string) (imageName, depName, fileNa
 		}
 		if ftype == pubsubFunc {
 			if rVersion.pubsubImage == "" {
-				err = errors.New("The given runtime and version does not have a valid image for event based functions")
+				var runtimeList []string
+				for i := range runtimeObjList {
+					for j := range runtimeObjList[i] {
+						if runtimeObjList[i][j].pubsubImage != "" {
+							runtimeList = append(runtimeList, runtimeObjList[i][j].runtimeID+runtimeObjList[i][j].version)
+						}
+					}
+				}
+				err = errors.New("The given runtime and version '" + runtime + "does not have a valid image for event based functions. Available runtimes are: " + strings.Join(runtimeList[:], ", "))
```

Comment:
Yes, it is better to have it in a function. I prefer it that way so once we need to update it is as simple as possible.